### PR TITLE
Fixes #12253 - notify is post to foreman succeeded, yet failed to sav…

### PR DESCRIPTION
**Pull request by shlomizadok**
…e in reports dir

@xprazak2 - can you please test?

### CODE REVIEW

**Comment by xprazak2** [comment]
Seems ok - no false positive logs about failures in forwarding to foreman. 
There is possibly one unpleasant scenario related to this. Correct me if I am wrong, but if error occurs while forwarding to foreman, exception gets raised [here](https://github.com/theforeman/smart_proxy_openscap/pull/20/files#diff-d9b04fc460aeec56309cdf7a9eb41a33R39). That means, report is not saved into reportsdir - which makes sense, because we do not have id returned from foreman. But when we try to send spool to foreman later and it succeeds, report gets deleted from [spooldir](https://github.com/theforeman/smart_proxy_openscap/blob/master/lib/smart_proxy_openscap/spool_forwarder.rb#L51-L53) but is not moved to reportsdir and we will not be able to request it later.

**Comment by shlomizadok** [comment]
@xprazak2, @ares, I will be adding a third directory `failed_reports` for reports who where posted to Foreman successfully but failed to save to the `reports` directory. Sounds reasonable? I can also add a rake task / cron job which tries to move `failed_reports` to `reports` directory. Let me know what you think.

**Comment by xprazak2** [comment]
@shlomizadok, That seems reasonable. In the comment above, I was worried mostly about reports that do not get posted to Foreman successfully right away and are forwarded later from spool. Do we have that covered or are you planning to handle it with new `failed_reports` as well?

**Comment by shlomizadok** [comment]
@xprazak2 - both :)

**Comment by xprazak2** [comment]
@shlomizadok, ok :-)

File: lib/smart_proxy_openscap/openscap_exception.rb [hunk 1]
```diff
@@ -2,5 +2,6 @@ module Proxy::OpenSCAP
   class OpenSCAPException < StandardError; end
   class StoreReportError < StandardError; end
   class StoreSpoolError <StandardError; end
+  class StoreFailedError <StandardError; end
```
Comment (xprazak2) [inline lib/smart_proxy_openscap/openscap_exception.rb]: nit: space after `<`

**Comment by xprazak2** [comment]
@shlomizadok: we probably want to add new `:failed_dir` into openscap.yml.example
The new storage seems to be working fine, but how do you run `smart-proxy-openscap-send` from source? When I point the script to the proxy by adding it to the `$LOAD_PATH` (replacing [this](https://github.com/theforeman/smart_proxy_openscap/blob/master/bin/smart-proxy-openscap-send#L12-L13)) it complaints about not finding stuff. What am I missing?

**Comment by shlomizadok** [comment]
@xprazak2 - I run it from irb, loading the `$LOAD_PATH` manually, and running `Proxy::OpenSCAP::send_spool_to_foreman` from irb

**Comment by shlomizadok** [comment]
@xprazak2 - Added `:failed_dir` into openscap.yml.example and fixed nitpicks

**Comment by xprazak2** [comment]
@shlomizadok, thx for the hint, it worked, and now it get:

NoMethodError: undefined method [ 'upload_path' ](https://github.com/theforeman/smart_proxy_openscap/blob/master/lib/smart_proxy_openscap/spool_forwarder.rb#L38) for #<Proxy::OpenSCAP::SpoolForwarder:0x0000000204dc40> 

It got probably moved to [ForemanForwarder](https://github.com/theforeman/smart_proxy_openscap/blob/master/lib/smart_proxy_openscap/foreman_forwarder.rb#L24-L26), so maybe extract it?

**Comment by shlomizadok** [comment]
@xprazak2 - fixed. Thanks!

**Comment by xprazak2** [comment]
@shlomizadok: I get another [error](http://pastebin.com/b40LtWFu) - sorry :cry:

Seems to me like openscap gem has a problem when parsing the report from spool, working fine when forwarding report directly from client. If this is only my local issue and you are able to send spool, I have no other comments.

**Comment by shlomizadok** [comment]
@xprazak2 - Thanks for finding this. Fixed (that was a tough one....)
Can you please retest so we can merge and package? Thanks :)

**Comment by xprazak2** [comment]
@shlomizadok: thx for fix, I have no problems while sending spool now, seems like we are ready to merge..

**Pull request merged**